app/assets/javascripts/controllers/container_dashboard/util/dashboard-utils-factory.js

```diff
@@ -97,8 +97,28 @@ angular.module('miq.util').factory('dashboardUtilsFactory', function() {
   var parseDate = function(date) {
     myDate = Date.parse(date);
     return isNaN(myDate) ? date : myDate
-  }
+  };
+  var iconClassForProvider = function(provider) {
```

Comment:
Agreed, it should eventually be more global.. I do actually have a little bit of progress in that direction in #5946, pushing the icon info into model decorators. But if we prefer that direction, then the logic should be on ruby side..